browser/src/core/Socket.js

```diff
@@ -1248,6 +1170,41 @@ app.definitions.Socket = L.Class.extend({
 		this._map.uiManager.showInfoModal('exported-success', _('Exported to storage'), _('Successfully exported: ') + decodeURIComponent(command.filename), '', _('OK'));
 	},
 
+	_showDocumentConflictPopUp: function() {
+		var buttonList = [];
+		var callbackList = [];
+
+		buttonList.push({ id: 'cancel-conflict-popup', text: _('Cancel') });
+		callbackList.push({ id: 'cancel-conflict-popup', func_: null });
+
+		buttonList.push({ id: 'discard-button', text: _('Discard') });
+		buttonList.push({ id: 'overwrite-button', text: _('Overwrite') });
+
+		callbackList.push({id: 'discard-button', func_: function() {
+			this.sendMessage('closedocument');
+		}.bind(this) });
+
+		callbackList.push({id: 'overwrite-button', func_: function() {
+			this.sendMessage('savetostorage force=1'); }.bind(this)
+		});
+
+		if (!this._map['wopi'].UserCanNotWriteRelative) {
+			buttonList.push({ id: 'save-to-new-file', text: _('Save to new file') });
+			callbackList.push({ id: 'save-to-new-file', func_: function() {
+				var filename = this._map['wopi'].BaseFileName;
+				if (filename) {
+					filename = L.LOUtil.generateNewFileName(filename, '_new');
+					this._map.saveAs(filename);
+				}
+			}.bind(this)});
+		}
+
+		var title = vex._escapeHtml(_('Document has been changed'));
+		var message = vex._escapeHtml(_('Document has been changed in storage. What would you like to do with your unsaved changes?'));
```

Comment:
_escapeHtml is not needed now, we don't build html  from strings

Reply:
Thanks for the comment :) Let me remove that forgotten part while i solve the conflict.